content/doc/pipeline/index.adoc

```diff
@@ -524,12 +525,29 @@ If you configured your pipeline to accept parameters using the *Build with
 Parameters* option, those parameters are accessible as Groovy variables of the
 same name.
 
+=== Types of Executors
+
+Every Pipeline build runs on a Jenkins master using a *flyweight executor*,
+which is an uncounted (because it's a temporary rather than configured) slot. Flyweight executors
+require very little computing power. A flyweight executor (sometimes also called
+a flyweight task) represents Groovy script, which is idle as it waits for a step to complete.
+
+To highlight the contrast between executor types, some Jenkins documentation calls any regular executor a heavyweight executor.
+
+When you run a `node` step, an executor is allocated on a node, which is usually an agent, as soon as
+an appropriate node is available. Avoid placing `input` within a node when you write pipeline syntax, because doing that ties up both the flyweight executor used for input and the regular executor used by the node block, which will not be free for other tasks until input is complete.
+
+The *Build Executor Status* widget on the Jenkins page displays status for both types of executors. If the
```

Comment:
A flyweight executor running a Pipeline build is hidden when the actual script is not doing anything (just inside some step).